slsim/lens.py

```diff
@@ -320,27 +337,56 @@ def image_observer_times(self, t_obs):
         observer_times = t_obs + arrival_times - np.min(arrival_times)
         return observer_times
 
-    def point_source_magnitude(self, band, lensed=False):
+    def point_source_magnitude(self, band, lensed=False, time=None):
         """Point source magnitude, either unlensed (single value) or lensed (array) with
         macro-model magnifications.
 
-        # TODO: time-variability with time-delayed and micro-lensing
+        # TODO: time-variability with micro-lensing
 
         :param band: imaging band
         :type band: string
         :param lensed: if True, returns the lensed magnified magnitude
         :type lensed: bool
+        :param time: time is a image observation time which is a astropy.unit object. If
+            None, provides magnitude without variability.
         :return: point source magnitude
         """
-        band_string = str("mag_" + band)
+        # band_string = str("mag_" + band)
         # TODO: might have to change conventions between extended and point source
-        source_mag = self._source_dict[band_string]
-        # TODO: requires time information and needs to be shifted for
-        # different arriving images
         if lensed:
-            mag = self.point_source_magnification()
-            return source_mag - 2.5 * np.log10(np.abs(mag))
-        return source_mag
+            magnif = self.point_source_magnification()
+            if time is not None:
+                if time.unit == u.day:
+                    time = time
+                else:
+                    time = time.to(u.day)
+                if self.source._variability_model is None:
+                    raise ValueError(
+                        "Variability model is not provided. Please choose"
+                        "one of the variability models from the Variability class."
+                    )
+                else:
+                    observed_time = []
+                    for t_obs in time.value:
+                        observed_time.append(self.image_observer_times(t_obs))
+                    transformed_observed_time = (
+                        np.array(observed_time).T.tolist() * u.day
+                    )
+                    variable_magnitude = self.source.magnitude(
+                        band,
+                        image_observation_times=transformed_observed_time,
+                    )
+                    magnif_log = 2.5 * np.log10(abs(magnif))
+                    lensed_variable_magnitude = (
+                        variable_magnitude - magnif_log[:, np.newaxis]
+                    )
+                    return lensed_variable_magnitude
+            else:
+                magnified_mag = self.source.magnitude(band) - 2.5 * np.log10(
```

Comment:
the magnification correction is effectively duplicated here. Try to make code such that each operation only appears on one place